96-chromebook.install

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/bash
+
+COMMAND="$1"
+KERNEL_VERSION="$2"
+BOOT_DIR_ABS="$3"
+KERNEL_IMAGE="$4"
+
+if [[ -d "$BOOT_DIR_ABS" ]]; then
+    INITRD="initrd"
+else
+    BOOT_DIR_ABS="/boot"
+    INITRD="initramfs-${KERNEL_VERSION}.img"
+fi
+
+case "$COMMAND" in
+    add)
+        echo "Install required packages"
+        dnf install lz4 vboot-utils uboot-tools -y
+        if [[ -f "$BOOT_DIR_ABS/$INITRD" ]]; then
+            rm -f -- "$BOOT_DIR_ABS/$INITRD"
+        fi
+        dracut --force -v --force-drivers "ulpi usb-storage phy-qcom-usb-hs-28nm \
+        phy-qcom-usb-ss ocmem dwc3 dwc3-of-simple dwc3-pci ehci-platform xhci-plat-hcd \
+        i2c-qcom-geni i2c-qup icc-osm-l3 qcom-spmi-pmic phy-qcom-qmp phy-qcom-qusb2 \
+        phy-qcom-usb-hs qcom_aoss qcom-apcs-ipc-mailbox llcc-qcom nvmem_qfprom smem \
+        smp2p dwc3-qcom \
+        " "$BOOT_DIR_ABS/$INITRD" --kver "$KERNEL_VERSION" --kmoddir /lib/modules/"${KERNEL_VERSION}"
+
+        kernel="${KERNEL_IMAGE##*/}-${KERNEL_VERSION}.lz4"
+        compression="lz4"
+        # Compress image
+        rm -f "$BOOT_DIR_ABS/${KERNEL_IMAGE##*/}-${KERNEL_VERSION}.lz4" || true
+        lz4 "$BOOT_DIR_ABS/${KERNEL_IMAGE##*/}-${KERNEL_VERSION}" "$BOOT_DIR_ABS/${KERNEL_IMAGE##*/}-${KERNEL_VERSION}.lz4"
+        dtbs=" \
+            -b /boot/dtb/qcom/sc7180-trogdor-coachz-r3.dtb \
+            -b /boot/dtb/qcom/sc7180-trogdor-lazor-r3-kb.dtb \
+            -b /boot/dtb/rockchip/rk3399-gru-kevin.dtb\
+            -b /boot/dtb/rockchip/rk3399-gru-scarlet-inx.dtb \
+            "
+        mkimage -D "-I dts -O dtb -p 2048" -i "$BOOT_DIR_ABS/$INITRD" \
+                -f auto -A arm64 -O linux -T kernel -C $compression -a 0 \
+                -d /boot/$kernel $dtbs \
+                kernel.itb
+
+	bootloader="boot_params"
+	echo "root=PARTUUID=%U/PARTNROFF=1 rootwait rw" > boot_params
```

Comment:
No, instead just use the cmdline in `/proc/cmdline`. Or better:

```
if [[ -f /etc/kernel/cmdline ]]; then
    read -r -d '' -a BOOT_OPTIONS < /etc/kernel/cmdline
elif [[ -f /usr/lib/kernel/cmdline ]]; then
    read -r -d '' -a BOOT_OPTIONS < /usr/lib/kernel/cmdline
else
    declare -a BOOT_OPTIONS

    read -r -d '' -a line < /proc/cmdline
    for i in "${line[@]}"; do
        [[ "${i#initrd=*}" != "$i" ]] && continue
        [[ "${i#BOOT_IMAGE=*}" != "$i" ]] && continue
        BOOT_OPTIONS+=("$i")
    done
fi
```

like is done by `/usr/lib/kernel/install.d/90-loaderentry.install`.

Reply:
noted